lib/repl.js

```diff
@@ -948,8 +999,10 @@ function complete(line, callback) {
     }
 
     completionGroupsLoaded();
-  } else if (match = line.match(requireRE)) {
-    // require('...<Tab>')
+  } else if (match = (line.match(requireRE) || line.match(fsRE))) {
+    // require('...<Tab>') and fs.<method>('...<Tab>')
+
+    const isFS = line.match(fsRE);
```

Comment:
This `RegExp` looks pretty intensive to me, we should not be running it twice. The `if` statement could be refactored to be `(matchRequire = line.match(requireRE)) || (matchFs = line.match(fsRE))`. Then we can assign `match` within the body.

(Of course, the `var` declaration will be needed for both.)

Reply:
definitely.